File: .skills/openclaw-skills/skills/liujuntao123/auto-free-banana/scripts/flow-webapi/exceptions.ts
export class FlowError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'FlowError';
  }
}

export class AuthError extends FlowError {
  constructor(message: string) {
    super(message);
    this.name = 'AuthError';
  }
}

export class TimeoutError extends FlowError {
  constructor(message: string) {
    super(message);
    this.name = 'TimeoutError';
  }
}

export class QuotaExceeded extends FlowError {
  constructor(message: string) {
    super(message);
    this.name = 'QuotaExceeded';
  }
}

export class GenerationError extends FlowError {
  constructor(message: string) {
    super(message);
    this.name = 'GenerationError';
  }
}

File: .skills/openclaw-skills/skills/neuhanli/smart-charts/core/exceptions.py
"""统一错误处理"""

from enum import Enum
from typing import Any, Dict, Optional


class ErrorCode(Enum):
    FILE_NOT_FOUND = 1001
    FILE_PERMISSION_DENIED = 1002
    FILE_FORMAT_INVALID = 1003
    FILE_SIZE_EXCEEDED = 1004
    DATA_PARSE_ERROR = 2001
    DATA_EMPTY = 2003
    DATA_TYPE_MISMATCH = 2004
    CHART_GENERATION_ERROR = 4001
    CHART_TYPE_UNSUPPORTED = 4002
    CHART_CONFIG_ERROR = 4003
    EXPORT_FORMAT_UNSUPPORTED = 6001
    EXPORT_FAILED = 6002
    TEMPLATE_NOT_FOUND = 7001
    TEMPLATE_FORMAT_UNSUPPORTED = 7002
    TEMPLATE_PARSE_ERROR = 7003
    UNKNOWN_ERROR = 9999


class SmartChartsError(Exception):

    def __init__(
        self,
        message: str,
        code: ErrorCode = ErrorCode.UNKNOWN_ERROR,
        details: Optional[Dict[str, Any]] = None,
    ):
        self.message = message
        self.code = code
        self.details = details or {}
        super().__init__(f"[{code.name}] {message}")

    def to_dict(self) -> Dict[str, Any]:
        return {
            "error": self.message,
            "code": self.code.value,
            "code_name": self.code.name,
            "details": self.details,
        }


class FileError(SmartChartsError):
    pass


class DataError(SmartChartsError):
    pass


class ChartError(SmartChartsError):
    pass


class ExportError(SmartChartsError):
    pass


class TemplateError(SmartChartsError):
    pass
